Realm/ObjectServerTests/RealmServer.swift

```diff
@@ -0,0 +1,780 @@
+        try? FileManager().removeItem(atPath: mongoUrl.appendingPathComponent(mongoDataDirectory).absoluteString)
+        try? FileManager().removeItem(atPath: "tmp")
+    }()
+
+    /// Launch the mongo server in the background.
+    /// This process should run until the test suite is complete.
+    private func launchMongoProcess() {
+        try? FileManager().createDirectory(atPath: mongoUrl.appendingPathComponent(mongoDataDirectory).absoluteString,
+                                           withIntermediateDirectories: false,
+                                           attributes: nil)
+
+        mongoProcess.launchPath = mongoUrl.appendingPathComponent("bin").appendingPathComponent("mongod").absoluteString
+        mongoProcess.arguments = [
+            "--quiet",
+            "--dbpath", "\(mongoUrl)/\(mongoDataDirectory)",
+            "--bind_ip", "localhost",
+            "--port", "26000",
+            "--replSet", "test"
+        ]
+        mongoProcess.standardOutput = nil
+        mongoProcess.launch()
+    }
+
+    private func launchServerProcess() {
+        let goRoot = rootUrl.appendingPathComponent("build").appendingPathComponent("go").absoluteString
+        let bundle = Bundle.init(for: RealmServer.self)
+        let serverBinary = bundle.path(forResource: "stitch_server", ofType: nil)
+        let createUserBinary = bundle.path(forResource: "create_user", ofType: nil)
+
+        // create the admin user
+        let userProcess = Process()
+        userProcess.environment = [
+            "GOROOT": goRoot,
+            "PATH": "$PATH:\(bundle.resourcePath!)",
+            "LD_LIBRARY_PATH":bundle.resourcePath!
+        ]
+        userProcess.launchPath = createUserBinary
+        userProcess.arguments = [
+            "addUser",
+            "-domainID",
+            "000000000000000000000000",
+            "-mongoURI", "mongodb://localhost:26000",
+            "-salt", "DQOWene1723baqD!_@#",
+            "-id", "unique_user@domain.com",
+            "-password", "password"
+        ]
+        userProcess.launch()
+        userProcess.waitUntilExit()
+
+        serverProcess.environment = [
+            "GOROOT": goRoot,
+            "PATH": "$PATH:\(bundle.resourcePath!)",
+            "LD_LIBRARY_PATH": bundle.resourcePath!
+        ]
+        // golang server needs a tmp directory
+        try? FileManager.default.createDirectory(atPath: "tmp", withIntermediateDirectories: false, attributes: nil)
+        serverProcess.launchPath = serverBinary
+        serverProcess.arguments = [
+            "--configFile",
+            bundle.path(forResource: "test_config", ofType: "json")!
+        ]
+
+        let pipe = Pipe()
+        pipe.fileHandleForReading.readabilityHandler = { file in
+            // prettify server output
+            let available = String(data: file.availableData, encoding: .utf8)?.split(separator: "\t")
+            print(available!.map { part -> String in
+                if part.contains("INFO") {
```

Comment:
Isn't this replacing the entire log line with just the emoji?

Reply:
No, stitch logs are separated by tabs, and the INFO/ERROR/DEBUG/etc. tags are standalone parts. So this will only replace the tag.